airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/SqlOperations.java

```diff
@@ -22,6 +22,18 @@
    */
   void createSchemaIfNotExists(JdbcDatabase database, String schemaName) throws Exception;
 
+  /**
+   * Denotes whether the schema exists in destination database
+   *
+   * @param database Database that the connector is syncing
+   * @param schemaName Name of schema.
+   *
+   * @return true if the schema exists in destination database, false if it doesn't
+   */
+  default boolean isSchemaExists(final JdbcDatabase database, final String schemaName) throws Exception {
```

Comment:
cc @sashaNeshcheret

Reply:
@sherifnada snowflake has a SQL statement IF NOT EXISTS and it was replaced by the changes. The cause for it is the 'create schema' statement needs 'CREATE SCHEMA', 'OWNERSHIP' or 'ALL' permissions on database level, and users with existing schema and 'Using' permissions on database level faced with issue `Could not connect with provided configuration. SQL access control error: Insufficient privileges to operate on database 'DATABASE`.
So instead of simply call `CREATE SCHEMA IF NOT EXISTS %s;` we firstly need to check it and try to create only if schema is absent. 
Regarding right place, i did it common interface because theoretically we can faced with same permissions behavior  in other db, for now i will move it from interface to snowflake operations impls.

Reply:
@sherifnada snowflake has a SQL statement IF NOT EXISTS and it was replaced by the changes. The cause for it is the 'create schema' statement needs 'CREATE SCHEMA', 'OWNERSHIP' or 'ALL' permissions on database level, and users with existing schema and 'Using' permissions on database level faced with issue `Could not connect with provided configuration. SQL access control error: Insufficient privileges to operate on database 'DATABASE`.
So instead of simply call `CREATE SCHEMA IF NOT EXISTS %s;` we firstly need to check it and try to create only if schema is absent. 
Regarding right place, i did it common interface because theoretically we can faced with same permissions behavior  in other db, for now i will move it from interface to snowflake operations impls.

Reply:
@sherifnada snowflake has a SQL statement IF NOT EXISTS and it was replaced by the changes. The cause for it is the 'create schema' statement needs 'CREATE SCHEMA', 'OWNERSHIP' or 'ALL' permissions on database level, and users with existing schema and 'Using' permissions on database level faced with issue `Could not connect with provided configuration. SQL access control error: Insufficient privileges to operate on database 'DATABASE`.
So instead of simply call `CREATE SCHEMA IF NOT EXISTS %s;` we firstly need to check it and try to create only if schema is absent. 
Regarding right place, i did it common interface because theoretically we can faced with same permissions behavior  in other db, for now i will move it from interface to snowflake operations impls.